build/IPDK_Container/docker/README_DOCKER.md

```diff
@@ -0,0 +1,223 @@
+
+### Commands to setup the example scenario:
+
+1) Startup a clean IPDK container daemon with './ipdk start -d'.
+
+2) Create 2 vhost ports using GNMI commands through the 'ipdk' script:
+    a. ./ipdk addintf --name net_vhost0 --host host1 --socket vhost-user-0 --type LINK
+    b. ./ipdk addintf --name net_vhost1 --host:host2 --socket vhost-user-1 --type LINK
+
+3) On your host, start your VMs using the following qemu commands or similar. 
+These enable your vhost ports within the VM to communicate with the container.
+- ./images/ubuntu.qcow2 and ./images/ubuntu2.qcow2 are the VM images
+- ./volume/ - Shared path with container
+
+VM1: 
+----
+qemu-kvm -smp 4 -m 1024M \
+    -boot c -cpu host -enable-kvm -nographic \
+    -L /root/pc-bios -name VM1_TAP_DEV \
+    -hda ./images/ubuntu.qcow2 \
+    -net nic -net user \
+    -object memory-backend-file,id=mem,size=1024M,mem-path=/mnt/huge,prealloc=yes,share=on \
+    -numa node,memdev=mem \
+    -chardev socket,id=char1,path=./volume/vhost-user-0 \
+    -netdev type=vhost-user,id=netdev0,chardev=char1,vhostforce \
+    -device virtio-net-pci,mac=00:e8:ca:11:aa:01,netdev=netdev0 \
+    -serial telnet::6551,server,nowait &
+
+sudo qemu-system-x86_64 -smp 4 -m 1024M \
+    -boot c -cpu host -enable-kvm \
+    -L /root/pc-bios -name VM1_TAP_DEV \
+    -hda ./images/ubuntu.qcow2 \
+    -net nic -net user \
+    -object memory-backend-file,id=mem,size=1024M,mem-path=/mnt/huge,prealloc=yes,share=on \
+    -numa node,memdev=mem \
+    -chardev socket,id=char1,path=./volume/vhost-user-0 \
+    -netdev type=vhost-user,id=netdev0,chardev=char1,vhostforce \
+    -device virtio-net-pci,mac=00:e8:ca:11:aa:01,netdev=netdev0 \
+    -serial telnet::6551,server,nowait &
+
+VM2:
+----
+qemu-kvm -smp 4 -m 1024M \
+    -boot c -cpu host -enable-kvm -nographic \
+    -L /root/pc-bios -name VM2_TAP_DEV \
+    -hda ./images/ubuntu2.qcow2 \
+    -net nic -net user \
+    -object memory-backend-file,id=mem,size=1024M,mem-path=/mnt/huge,prealloc=yes,share=on \
+    -numa node,memdev=mem \
+    -chardev socket,id=char2,path=./volume/vhost-user-1 \
+    -netdev type=vhost-user,id=netdev1,chardev=char2,vhostforce \
+    -device virtio-net-pci,mac=00:e8:ca:11:bb:01,netdev=netdev1 \
+    -serial telnet::6552,server,nowait &
+
+sudo qemu-system-x86_64 -smp 4 -m 1024M \
+    -boot c -cpu host -enable-kvm \
+    -L /root/pc-bios -name VM2_TAP_DEV \
+    -hda ./images/ubuntu2.qcow2 \
+    -net nic -net user \
+    -object memory-backend-file,id=mem,size=1024M,mem-path=/mnt/huge,prealloc=yes,share=on \
+    -numa node,memdev=mem \
+    -chardev socket,id=char2,path=./volume/vhost-user-1 \
+    -netdev type=vhost-user,id=netdev1,chardev=char2,vhostforce \
+    -device virtio-net-pci,mac=00:e8:ca:11:bb:01,netdev=netdev1 \
+    -serial telnet::6552,server,nowait &
+
+
+4) On VMs: Configure IP-Address and static ARP/route on the vhost interface for both VMs. 
+Following is an example:
+
+VM1:
+----
+ifconfig eth0 1.1.1.1/24 up
```

Comment:
The [rundemo.sh](https://github.com/ipdk-io/ipdk/blob/main/build/IPDK_Container/scripts/rundemo.sh) script does this automatically using cloud-init. I suggest you just use that rather than having these commands here, as it's much more error prone to have to enter them manually.